File: Week5/Day5/Solutions/C/prog2.c
int printf(const char*,...);
int scanf(const char*,...);

#include<string.h>

void main() {
	char hex[5];
	scanf("%s",hex);
	int len = strlen(hex);
	int base = 1,dec =0,bin[10],i=0,rem,cnt=0;
	for(int i = len-1;i>=0;i--){
		if(hex[i]>='0'&& hex[i]<='9'){
			dec += (hex[i]-48)*base;
		}	
		else if(hex[i]>='A'&& hex[i]<='F'){
			dec += (hex[i]-55)*base;
		}	
		base *=  16;
	}//Decimal number received in dec

	while(dec){
		rem=dec%2;
		dec/=2;
		bin[i] = rem;
		cnt+=1;	
		i++;
	}//Binary number received in bin[]

	printf("Decimal Number:");
	for(int j=cnt-1;j>=0;j--)
		printf("%d",bin[j]);
}

File: Week6/Day6/Solutions/Python/prog2.py
print("------Menu-----\n1)Binary\n2)Hexadecimal\n3)Octal")

num = int(input("\nEnter number:"))
length = len(str(num))
choice = int(input("Enter choice: "))

def converter(num,length,choice):
    while num:
        rem = num//10**(length-1)
        num%=10**(length-1)
        length-=1
        if choice == 1:
            print("Binary of",rem,"=",str(bin(rem))[2:])
        elif choice == 2:
            print("Hexadecimal of",rem,"=",str(hex(rem))[2:])
        elif choice == 3:
            print("Octal of",rem,"=",str(oct(rem))[2:])

converter(num,length,choice)
